lib/std/zig/system/darwin/macos.zig

```diff
@@ -74,20 +74,40 @@ pub fn detect(target_os: *Target.Os) !void {
     return error.OSVersionDetectionFail;
 }
 
-fn parseSystemVersion(buf: []const u8) !std.builtin.Version {
+fn parseSystemVersion(buf: []const u8) !std.SemanticVersion {
     var svt = SystemVersionTokenizer{ .bytes = buf };
     try svt.skipUntilTag(.start, "dict");
     while (true) {
         try svt.skipUntilTag(.start, "key");
         const content = try svt.expectContent();
         try svt.skipUntilTag(.end, "key");
-        if (std.mem.eql(u8, content, "ProductVersion")) break;
+        if (mem.eql(u8, content, "ProductVersion")) break;
     }
     try svt.skipUntilTag(.start, "string");
     const ver = try svt.expectContent();
     try svt.skipUntilTag(.end, "string");
 
-    return std.builtin.Version.parse(ver);
+    const parseVersionComponent = struct {
+        fn parseVersionComponent(component: []const u8) !usize {
+            return std.fmt.parseUnsigned(usize, component, 10) catch |err| {
+                switch (err) {
+                    error.InvalidCharacter => return error.InvalidVersion,
+                    error.Overflow => return error.Overflow,
+                }
+            };
+        }
+    }.parseVersionComponent;
+
+    var version_components = mem.split(u8, ver, ".");
+    const major = version_components.first();
+    if (major.len == 0) return error.InvalidVersion;
+    const minor = version_components.next() orelse return error.InvalidVersion;
+    const patch = version_components.next() orelse "0";
+    return .{
+        .major = try parseVersionComponent(major),
+        .minor = try parseVersionComponent(minor),
+        .patch = try parseVersionComponent(patch),
+    };
```

Comment:
@kubkon do you think a version string like this is possible?
```      
      \\	<string>14</string>
```
Currently the code assumes a version is either in the format of X.X or X.X.X and only allows the patch to not be specified in which case it defaults to 0.

And other than do you think this looks ok to parse the versions in that XML?

Reply:
I believe Apple's products always contain major and minor version in the plist, therefore at a minimum I would expect `14.0` or similar. It also seems to agree with our assumptions in the opening comments https://github.com/ziglang/zig/blob/f211c1559ab8c25a52d45e495ef42d4937e98e97/lib/std/zig/system/darwin/macos.zig#L13

Reply:
Alright, looks like this would work then.